contentScript.js

```diff
@@ -0,0 +1,128 @@
+(() => {
+  let ytLeftControls, ytPlayer;
+  let currentVideo = null; // id of the currently playing video
+  let lastVideo = null; // id of the last handled video
+  let bookmarksList = []; // list of the video bookmarks
+
+  /* Fetch all bookmarks for current video */
+  const fetchBookmarks = () => {
+    return new Promise((resolve) => {
+      chrome.storage.sync.get([currentVideo], (obj) => {
+        resolve(obj[currentVideo] ? JSON.parse(obj[currentVideo]) : []);
+      });
+    });
+  };
+
+  /* get current video's duration */
+  const getVideoDuration = () => {
+    if (ytPlayer) return parseInt(ytPlayer.duration);
+    return null;
+  };
+
+  /* Handle request to add new bookmark */
+  const addNewBookmarkEventHandler = () => {
+    let currentTime = null;
+    if (ytPlayer) currentTime = parseInt(ytPlayer.currentTime);
+
+    if (currentTime) {
+      const newBookmark = {
+        time: currentTime,
+        desc: "Bookmark at " + getTime(currentTime),
+      };
+
+      fetchBookmarks().then((obj) => {
+        bookmarksList = obj;
+        let newBookmarksList = bookmarksList.concat(newBookmark);
+        newBookmarksList = newBookmarksList.sort((a, b) => a.time - b.time);
+        chrome.storage.sync.set({
+          [currentVideo]: JSON.stringify(newBookmarksList),
+        });
+      });
+    }
+  };
+
+  /* this checks for whenever a new video is loaded or the current tab is reloaded */
+  const newVideoLoaded = (refreshed) => {
+    bookmarksList = [];
+    ytLeftControls = document.getElementsByClassName("ytp-left-controls")[0];
+    ytPlayer = document.getElementsByClassName("html5-main-video")[0];
+
+    const fetchBookmarksPromise = fetchBookmarks();
```

Comment:
why not await it here instead of using Promise.all that wait you don't have to index through the Promise results the way you do on line 65